lib/BUILD

```diff
@@ -13,6 +13,9 @@ objc_library(
     srcs = ["swizzle_absolute_xcttestsourcelocation.m"],
     visibility = ["//visibility:public"],
     alwayslink = True,
+    target_compatible_with = [
+        "@platforms//os:macos",
```

Comment:
does this work for iOS still, since the target isn't macos?

Reply:
This is similar to what I had to put into rules_swift. Seems like we really need a "is apple platform" check.

Reply:
Assuming you mean [this](https://github.com/bazelbuild/rules_swift/pull/878/files), how is that different if we were to do this here? What's the preferred approach?
```
target_compatible_with = [
    "@platforms//os:macos",
    "@platforms//os:watchos",
    "@platforms//os:ios",
    "@platforms//os:tvos",
]
```

Reply:
Did that way not work?

Reply:
So I went the same way that rules_swift did in the beginning, but that seems to break on non-macOS platforms (in https://github.com/bazelbuild/bazel-central-registry/pull/196): https://buildkite.com/bazel/bcr-presubmit/builds/541#0183a163-1ec0-4329-b265-8aca36c17d33 

Which makes me wonder if this is working in rules_swift at all. I'll try to go with the `platforms` solution instead now.

Reply:
The Linux CI started to pass after we did it that way, so I'm pretty sure it's working there.

Reply:
Reverted this to use platforms and it seems to work in the central-registry too: https://github.com/bazelbuild/bazel-central-registry/pull/196